Src/Newtonsoft.Json/JsonConverter.cs

```diff
@@ -96,4 +97,75 @@ public virtual bool CanWrite
             get { return true; }
         }
     }
+
+    /// <summary>
+    /// Converts an object to and from JSON.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
+    public abstract class JsonConverter<T> : JsonConverter
+    {
+        /// <summary>
+        /// Writes the JSON representation of the object.
+        /// </summary>
+        /// <param name="writer">The <see cref="JsonWriter"/> to write to.</param>
+        /// <param name="value">The value.</param>
+        /// <param name="serializer">The calling serializer.</param>
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
```

Comment:
I considered that but it would be a breaking change since I changed existing non-sealed JsonConverter classes' base class to this new generic JsonConverter.